test/wasi/test-wasi.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+require('../common');
+
+const cp = require('child_process');
+const fs = require('fs');
+const path = require('path');
+const assert = require('assert');
+
+const SKIP = [
+  'poll',
+];
+
+const C_DIR = path.resolve(__dirname, 'c');
+const WASM_DIR = path.resolve(__dirname, 'wasm');
+
+function parseOptions(string) {
+  const options = {};
+  for (const line of string.split('\n')) {
+    if (!line.startsWith('//')) {
+      break;
+    }
+    const [key, value] = line.slice(3).split(': ');
+    if (!key || !value) {
+      continue;
+    }
+    options[key] = value.replace(/\\n/g, '\n');
+  }
+  return options;
+}
+
+function exec(command, { stdin } = {}) {
+  try {
+    const stdout = cp.execSync(command, {
+      input: stdin,
+    });
+    return { code: 0, stdout: stdout.toString() };
+  } catch (e) {
+    if (e.status) {
+      return { code: e.status, stdout: '' };
+    }
+    throw e;
+  }
+}
+
+const ARGS = [
+  '--expose-internals',
+  `-r ${__dirname}/patch_preopen`,
```

Comment:
Is there a particular reason that you have to write a new test runner here? Are the `.c` sources copied from somewhere else so that we have to adapt to them? It would be easier to debug if we just write different `test-*.js` files and put the options there, and we can also reuse the existing status file management from the python test runner that way.

The only exception that we have is WPT but that's because the tests are maintained in a separate repo. Is that   also the case for these tests here? If so we also need some comments in the README (on a side note, it would be nice if there is a common WASI test suite)

Reply:
the tests here are copied from lucet, and yeah there has been some discussion on irc of getting a central wasi test suite.

Reply:
If we can't touch the `.c`, I think we can still create something similar to `require('../common/wpt').WPTRunner` with what you have in the callback of `readdirSync` here, and use that to drive the tests in separate files so that it's possible to run individual tests.

Reply:
(But anyway this is not blocking,  I think this part can land as long as it actually runs in the CI)